Lib/enum.py

```diff
@@ -568,10 +568,11 @@ def __new__(metacls, cls, bases, classdict, *, boundary=None, _simple=False, **k
         try:
             exc = None
             enum_class = super().__new__(metacls, cls, bases, classdict, **kwds)
-        except RuntimeError as e:
-            # any exceptions raised by member.__new__ will get converted to a
-            # RuntimeError, so get that original exception back and raise it instead
-            exc = e.__cause__ or e
+        except Exception as e:
+            # since 3.12 the line "Error calling __set_name__ on '_proto_member' instance ..."
+            # is tacked on to the error instead of raising a RuntimeError
+            # recreate the exception to discard
+            exc = type(e)(str(e))
```

Comment:
The type and error message of the original exception are retained, but the stack trace and other properties of the original exception are lost.

Reply:
The stack trace would be useful.  What other properties are you thinking of?